test/dashboard.e2e.js

```diff
@@ -1,26 +1,108 @@
-const { startApp, stopApp } = require('./utils')
-
-// Skipping because the app opens the About window before the main Window
-// which is picked by spectron to query for dashboard elements.
-// This needs to be enabled after carefully fixing the autoupdate code that
-// causes this behaviour in NODE_ENV='test' in `main/index.js:170`
-describe.skip('Dashboard', () => {
+const { startApp, stopApp, screenshotApp } = require('./utils')
+import En from '../lang/en.json'
 
+describe('Dashboard tests', () => {
   let app
 
   beforeAll(async () => {
-    jest.setTimeout(30000)
-    // Launch the app
     app = await startApp()
   })
 
   afterAll(async () => {
     await stopApp(app)
   })
 
-  it('5 test cards are visible', async () => {
-    await expect(app.client
-      .isVisible('div[data-testid="card"]')
-    ).resolves.toHaveProperty('length', 5)
+  test('Run button is displayed on Dashboard', async () => {
+    const runButtonText = await app.client
+      .$('button[data-testid=button-dashboard-run]')
+      .getText()
+
+    expect(runButtonText).toMatch('Run')
+  })
+
+  describe('All 5 test cards are visible', () => {
+    const testDetails = [
+      {
+        id: 'websites',
+        name: 'Websites',
+        desc: 'Test the blocking of websites',
+      },
+      {
+        id: 'im',
+        name: 'Instant Messaging',
+        desc: 'Test the blocking of instant messaging apps',
+      },
+      {
+        id: 'circumvention',
+        name: 'Circumvention',
+        desc: 'Test the blocking of censorship circumvention tools',
+      },
+      {
+        id: 'performance',
+        name: 'Performance',
+        desc: 'Test your network speed and performance',
+      },
+      {
+        id: 'middlebox',
+        name: 'Middleboxes',
+        desc: 'Detect middleboxes in your network',
+      },
+    ]
+
+    testDetails.forEach((itr) => {
```

Comment:
Is this the same as using the [`test.each`](https://jestjs.io/docs/api#testeachtablename-fn-timeout) API from Jest?

Reply:
Yes, it is